nanogpt/basic_tokenizer.h

```diff
@@ -0,0 +1,115 @@
+#include <iostream>
+#include <fstream>
+#include <vector>
+#include <string>
+#include <unordered_map>
+#include <sstream>
+
+class BasicTokenizer {
+public:
+    BasicTokenizer(const std::string& filePath) {
+        std::ifstream file(filePath);
+
+        if (!file) {
+            std::cerr << "Unable to open file";
+            exit(9); // return with error code
+        }
+        std::string str((std::istreambuf_iterator<char>(file)), std::istreambuf_iterator<char>());
+
+        // Remove the first and last braces
+        str[0] = ' ';
+        str.pop_back();
+
+        std::stringstream ss(str);
+        std::string kv_pair;
+        while (std::getline(ss, kv_pair, ',')) {
```

Comment:
I'm worried that this file is too complicated for our use case. We could simplify things a lot by pre-processing this .json file to make it easier to parse in C++.

Should be able to write a small python script that does:
- For each entry in the json dict
  - Write `<key string encoded as UTF-8><TAB character><token value as ASCII digits><newline>`

I checked the .json file and didn't see any tab characters, so that should be ok to use as a separator. If that's ambiguous, then we could write out a binary file like
`<little-endian uint32_t: number of UTF-8 bytes in the string><UTF-8 bytes...><little-endian uint32_t:token value>` with no newlines.

Then we should be able to read each line of the file and parse it with sscanf() into two strings, and convert the second string into an integer.

Reply:
Here's a demo of what I'm thinking: https://github.com/dbort/et-tutorials/pull/4

It's not really shorter, but it might be easier to understand for users.